### DISCUSSION TITLE
Is it normal for Project Id to be part of the sign-in url of an email auth link.

### DISCUSSION THREAD

**Discussion by tomskopek** [invertase/react-native-firebase#6758] [Q&A]
Hi!

I posted a [much longer question](https://github.com/invertase/react-native-firebase/discussions/6730) earlier this week. I'm just looking for a quick sanity check:

Is it normal for my **project id** to be included in the url that I receive from a sign-in link?

The url received in the Email:

```
https://pdpip.er/?link=https://piedpiper-c3xa9.firebaseapp.com/__/auth/action?apiKey%SOME_API_KEY%26mode%3DsignIn%26oobCode%3DSOME_OOB_CODE%26continueUrl%3Dhttps://piedpiper.com/auth/confirm-email?email%253Dtom%2540gmail.com%26lang%3Den&apn=com.piedpiper&amv=1.66.2&ibi=com.piedpiper.ios&ifl=https://piedpiper-c3xa9.firebaseapp.com/__/auth/action?apiKey%SOME_API_KEY%26mode%3DsignIn%26oobCode%3DSOME_OOB_CODE%26continueUrl%3Dhttps://piedpiper.com/auth/confirm-email?email%253Dtom%2540gmail.com%26lang%3Den
```

Breaking this huge url down:

`https://pdpip.er/?link=`
So far so good, this looks like my regular dynamic links

...`https://piedpiper-c3xa9.firebaseapp.com/__/auth/action` <-------------
Uh oh, what's my firebase project id doing here? Is that normal?      

...`apiKey%SOME_API_KEY%26mode%3DsignIn%26oobCode%3DSOME_OOB_CODE%26`
Bunch of important stuff

...`continueUrl%3Dhttps://piedpiper.com/auth/confirm-email?email%253Dtom%2540gmail.com` <--------------
This is the deepLink I want! This is where I'll be navigating to in my app

...`%26lang%3Den&apn=com.piedpiper&amv=1.66.2&ibi=com.piedpiper.ios&`
This looks like some iOS specific stuff

...`ifl=https://piedpiper-c3xa9.firebaseapp.com/__/auth/action......`
The rest is simply repeating

This is my project settings:
![CleanShot 2022-12-01 at 14 24 26@2x](https://user-images.githubusercontent.com/4926143/205142028-0969ce7b-4baf-410d-a669-6c4772dcc58f.png)

Currently I'm doing a big hack on the receiving end to parse the continueUrl out, then append the API and OobCode query params back on. And the behaviour when connected to the firebase emulator is completely different, so I need to handle that differently. I've got it working but it's quite horrible.

**Comment by mikehardy** [comment]
Apologies - this isn't a feature I use personally so I'm not that familiar with it, and no one else appears available to answer questions. It's not really react-native-firebase specific though - I encourage you to post on stackoverflow where the upstream Firebase team offers support if you tag up the question well

**Reply by tomskopek** [reply]
Thanks @mikehardy 

In the last hour or so I found this part of the docs which shed some light on this: https://firebase.google.com/docs/auth/custom-email-handler. Wish I had read that part sooner.

It's starting to look like this is the expected behaviour/format for these links, and it's actually my link listener's pattern which is not standard. Will have to think about a better pattern to handle these incoming links.

![CleanShot 2022-12-01 at 16 27 40@2x](https://user-images.githubusercontent.com/4926143/205163192-3c95fa98-6ed2-4a58-a8d8-e7a9767552ef.png)

It's still quite puzzling to me how/why the firebase auth emulator creates such differently formatted links. But at least I feel like I've made some progress.
